### PR TITLE
CP-28152 add output option to planex-clone

### PR DESCRIPTION
**Pull request by samuelconnolly**
Used to state where to put clone_json if alternate location would be
preferred, also defaults to . so it writes config file to pwd

### CODE REVIEW

File: planex/cmd/clone.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ def parse_args_or_exit(argv=None):
                            "link the sources and the patchqueue and use "
                            "guilt to apply all the additional patches "
                            "[experimental]")
+    parser.add_argument("--output", "-o", default=".",
```
Comment (srowe) [inline planex/cmd/clone.py]: I think an output filename rather than a directory would be clearer.

Comment (samuelconnolly) [reply planex/cmd/clone.py]: I was going for a fixed filename since I used a fixed one in the build scripts admittedly.

Comment (srowe) [reply planex/cmd/clone.py]: As the caller consumes the file it better for that to define the entire filename.

File: planex/cmd/clone.py [hunk 2]
```diff
@@ -165,7 +167,7 @@ def clone_all(args, pin):
                  "name but different commitish is not supported.")
 
     if args.jenkins:
-        clone_jenkins(gathered)
+        clone_jenkins(args, gathered)
```
Comment (srowe) [inline planex/cmd/clone.py]: Only `output` is needed to pass in `args.output`

File: planex/cmd/clone.py [hunk 3]
```diff
@@ -42,6 +42,8 @@ def parse_args_or_exit(argv=None):
                            "link the sources and the patchqueue and use "
                            "guilt to apply all the additional patches "
                            "[experimental]")
+    parser.add_argument("--output", "-o", default="./clone_sources",
```
Comment (srowe) [inline planex/cmd/clone.py]: Can the default have a `.json` suffix, and the leading `./` is unnecessary.

File: planex/cmd/clone.py [hunk 4]
```diff
@@ -55,12 +57,12 @@ def repo_name(url):
...
 
 
-def clone_jenkins(gathered):
+def clone_jenkins(filename, gathered):
     """Print json file containing repositories to clone"""
     json_dict = {}
     for url, commitish in gathered:
         json_dict[repo_name(url)] = {'URL': url, 'commitish': commitish}
-    with open("clone_sources.json", "w+") as clone_sources:
+    with open(filename, "w+") as clone_sources:
```
Comment (srowe) [inline planex/cmd/clone.py]: Just `"w"` should be sufficient.

**Pull request merged**